### PR TITLE
AM - fix: add additional properties support

### PR DESCRIPTION
**Pull request by akashrpo**
## Description of the change

> add addl. Properties Support (productID, country, city, Latitude, Longitute, region, dma)

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by yashasvibajpai** [comment]
LGTM👍

**Review by abandonned — DISMISSED** [review]

File: v0/destinations/am/transform.js [hunk 1]
```diff
@@ -73,6 +73,7 @@ function setPriceQuanityInPayload(message, rawPayload) {
 }
 
 function createRevenuePayload(message, rawPayload) {
+  rawPayload.productId = message.properties.product_id;
```
Comment (saikumarrs) [inline v0/destinations/am/transform.js:76]: This is already part of the revenue mapping JSON.

File: v0/destinations/am/transform.js [hunk 2]
```diff
@@ -206,6 +207,8 @@ function responseBuilderSimple(
       endpoint = ENDPOINT;
       // event_type for identify event is $identify
       rawPayload.event_type = EventType.IDENTIFY_AM;
+      rawPayload.country = get(message, "context.location.country");
+      rawPayload.city = get(message, "context.location.city");
```
Comment (saikumarrs) [inline v0/destinations/am/transform.js:211]: Some of the mapping changes made here can be driven from the mapping config JSON files

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: v0/destinations/am/transform.js (hunk 2 above)
Comment (akashrpo) [reply v0/destinations/am/transform.js:211]: changed to use default mapping config JSON

File: v0/destinations/am/transform.js (hunk 1 above)
Comment (akashrpo) [reply v0/destinations/am/transform.js:76]: revenue mapping JSON is being used when `revenue` & `revenue_type` is being provided in properties. In this case, when we simply provide `product_id` we are passing that info in the payload.

File: v0/destinations/am/transform.js (hunk 2 above)
Comment (saikumarrs) [inline v0/destinations/am/transform.js:211]: Let's not take `country` and `city` fields from `context.location` as they are not part of our common API spec.

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**